module_userinfo.lua

```diff
@@ -0,0 +1,164 @@
+-- Copyright (C) 2022 Mjöllnir#3515
+-- This file is part of the "Not a Bot" application
+-- For conditions of distribution and use, see copyright notice in LICENSE
+
+local Date = Discordia.Date
+
+Module.Name = "userinfo"
+
+
+-- We have to precede special chars with an \ to prevent discord from replacing them with the corresponding emoji :<color>_circle:
+local discordStatus = { online = "\\🟢 Online", dnd = "\\🔴 Do Not Disturb", idle = "\\🟡 Idle", offline = "\\⚪ Offline" }
+local DEFAULT_COLOR = 0 -- Default color value, 0 == black
+local JOIN_ORDER_WINDOW = 7 -- Number of members to show in "Join order" field
+
+local sortedRolesCache = {}
```

Comment:
You're using a global cache for all guilds, this isn't a good idea.
Also, roles are already cached, there's no need to cache them.

Reply:
If you're unsure about what you should do with the cache, remove it, I'll add it myself once merged.

Reply:
> Also, roles are already cached, there's no need to cache them.
As we have seen above, only sorted roles tables are stored in this cache, not the roles given by Member.roles
> If you're unsure about what you should do with the cache, remove it, I'll add it myself once merged.
yep, maybe it's better because I don't see how to do that for the moment, I'd have to study the bot and discordia more in depth, but it might take some time